js/src/tooltip.js

```diff
@@ -254,8 +254,9 @@ const Tooltip = (($) => {
       if (this.isWithContent() && this._isEnabled) {
         $(this.element).trigger(showEvent)
 
+        const shadowRoot = Util.findShadowRoot(this.element)
         const isInTheDom = $.contains(
-          this.element.ownerDocument.documentElement,
+          shadowRoot !== null ? shadowRoot : this.element.ownerDocument.documentElement,
```

Comment:
This code is trying to find the "root" `Element`, either `document.documentElement` or a `ShadowRoot`, but it is split into two pieces: the `findShadowRoot` function, which may return `null`, and this ternary, which will handle `null`. Why not just fold these two pieces into one function whose purpose is to find the root element?

```js
// Returns the root Element, either document.documentElement or a ShadowRoot.
findRootElement (element) {
  // Go straight to the root node if the browser supports this method.
  if (typeof element.getRootNode === 'function') {
    const node = element.getRootNode()
    return (node instanceof ShadowRoot) ? node : node.documentElement
  }

  // Walk the parent chain to the root node.
  while (element.parentNode) {
    // Check for shadowRoot on the way up.
    if (element instanceof ShadowRoot) {
      return element
    }
    element = element.parentNode
  }
  // At this point, element has no parentNode. Thus, it is the root element.

  return element
}
```

Reply:
I agree that's better like the way you suggested 👍